src/blocks/timewarrior.rs

```diff
@@ -0,0 +1,227 @@
+        widget.set_values(values);
+        api.set_widget(widget).await?;
+
+        select! {
+            _ = sleep(config.interval.0) => (),
+            event = api.event() => match event {
+                UpdateRequest => (),
+                Action(a) => {
+                    if a == "stop_continue" {
+                        stop_continue().await?;
+                    }
+                },
+            }
+        }
+    }
+}
+
+/// Raw output from timew
+#[derive(Deserialize, Debug)]
+struct TimewarriorRAW {
+    pub id: u32,
+    pub start: String,
+    pub tags: Vec<String>,
+    pub annotation: Option<String>,
+    pub end: Option<String>,
+}
+
+/// TimeWarrior entry
+#[derive(Debug, PartialEq)]
+struct TimewarriorData {
+    pub id: u32,
+    pub start: DateTime<chrono::offset::Utc>,
+    pub tags: Vec<String>,
+    pub annotation: Option<String>,
+    pub end: Option<DateTime<chrono::offset::Utc>>,
+}
+
+impl From<TimewarriorRAW> for TimewarriorData {
+    fn from(item: TimewarriorRAW) -> Self {
+        Self {
+            id: item.id,
+            tags: item.tags,
+            annotation: item.annotation,
+            start: DateTime::from_utc(
+                chrono::NaiveDateTime::parse_from_str(&item.start, "%Y%m%dT%H%M%SZ").unwrap(),
+                chrono::Utc,
+            ),
+            end: item.end.map(|v| {
+                DateTime::from_utc(
+                    chrono::NaiveDateTime::parse_from_str(&v, "%Y%m%dT%H%M%SZ").unwrap(),
+                    chrono::Utc,
+                )
+            }),
+        }
+    }
+}
+
+/// Format a DateTime given a format string
+#[allow(dead_code)]
+fn format_datetime(date: &DateTime<chrono::Utc>, format: &str) -> String {
+    date.format(format).to_string()
+}
+
+/// Execute "timew export now" and return the result
+async fn call_timewarrior() -> Result<String> {
+    let out = Command::new("timew")
+        .args(["export", "now"])
+        .output()
+        .await
+        .error("failed to run timewarrior")?
+        .stdout;
+    let output = std::str::from_utf8(&out)
+        .error("failed to read output from timewarrior (invalid UTF-8)")?
+        .trim();
+    Ok(output.to_string())
+}
+
+/// Stop or continue a task
+async fn stop_continue() -> Result<()> {
+    let mut execute_continue: bool = true;
+    if let Some(tw) = process_timewarrior_data(&call_timewarrior().await?) {
+        // we only execute continue if the current task is stopped
+        // i.e. has an end defined
+        execute_continue = tw.end.is_some();
+    }
+
+    // is there a more rust way of doing this?
+    let args = match execute_continue {
+        true => "continue",
+        false => "stop",
+    };
+
+    Command::new("timew")
+        .args([args])
+        .stdout(std::process::Stdio::null())
+        .spawn()
+        .error("Error spawning timew")?
+        .wait()
+        .await
+        .error("Error executing stop/continue")
+        .map(|_| ())
```

Comment:
It might make sense to check `ExitStatus::success()`

Reply:
By how it's being called I think it should always return 0 but I guess there's nothing wrong with checking and returning an Err if it fails